packages/e2e/test/k6/endpoints/stake-pool/search.test.js

```diff
@@ -0,0 +1,56 @@
+import { check } from 'k6';
+import http from 'k6/http';
+
+const STAKE_POOL_PROVIDER_URL = __ENV.PROVIDER_SERVER_URL;
+const MAX_STAKE_POOLS = 500;
+export const options = {
+  thresholds: {
+    http_req_duration: ['p(95)<500'],
+    http_req_failed: ['rate<0.01']
+  }
+};
+
+export const setup = () => {
+  let stakePools = [];
+
+  // Get the list of stake pools
+  while (stakePools.length < MAX_STAKE_POOLS) {
+    const searchResponse = http.post(
+      `${STAKE_POOL_PROVIDER_URL}/stake-pool/search`,
+      JSON.stringify({
+        pagination: {
+          limit: 25, // Adjust the limit as needed
+          startAt: 0
+        }
+      }),
+      {
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      }
+    );
+    check(searchResponse, {
+      'Initial query successfull': (r) => r.status === 200
+    });
+    const parsedPageResults = JSON.parse(searchResponse.body).pageResults;
+    stakePools = [...stakePools, ...parsedPageResults];
```

Comment:
```suggestion
    stakePools.push(...parsedPageResults);
```

is equivalent but avoids to allocate a new `Array` and _copy all the elements_ at each iteration.

Reply:
Is this ready for re-review @iadmytro?